models/review.go

```diff
@@ -373,36 +423,53 @@ func SubmitReview(doer *User, issue *Issue, reviewType ReviewType, content, comm
 		return nil, nil, err
 	}
 
+	// try to remove team review request if need
+	if issue.Repo.Owner.IsOrganization() && (reviewType == ReviewTypeApprove || reviewType == ReviewTypeReject) {
+		teamReviewRequests := make([]*Review, 0, 10)
+		if err = sess.SQL("SELECT * FROM review WHERE reviewer_team_id > 0 AND type = ?", ReviewTypeRequest).Find(&teamReviewRequests); err != nil {
+			return nil, nil, err
+		}
+
+		if len(teamReviewRequests) > 0 {
+			for _, teamReviewRequest := range teamReviewRequests {
+				ok := false
+				if ok, err = isTeamMember(sess, issue.Repo.OwnerID, teamReviewRequest.ReviewerTeamID, doer.ID); err != nil {
+					return nil, nil, err
+				}
+
+				if ok {
+					if _, err := sess.Delete(teamReviewRequest); err != nil {
+						return nil, nil, err
+					}
+				}
+			}
+		}
+	}
+
 	comm.Review = review
 	return review, comm, sess.Commit()
 }
 
 // GetReviewersByIssueID gets the latest review of each reviewer for a pull request
-func GetReviewersByIssueID(issueID int64) (reviews []*Review, err error) {
-	reviewsUnfiltered := []*Review{}
+func GetReviewersByIssueID(issueID int64) ([]*Review, error) {
```

Comment:
This function appears to have changed considerably.

Why aren't you loading the reviewers anymore?

And we should still do both queries in the same transaction to prevent getting conflicting data.

Reply:
I think it's brtter, load it before use it by ``reviewer.LoadReviewer()`` is enough.

Reply:
> And we should still do both queries in the same transaction to prevent getting conflicting data.

Hmm, I'm sorry but I can't fully understand this issue, Would you please give me more detail abut it? Thanks.

Reply:
>> And we should still do both queries in the same transaction to prevent getting conflicting data.

> Hmm, I'm sorry but I can't fully understand this issue, Would you please give me more detail abut it? Thanks.

The follow two db queries should happen within a transaction (`sess.Begin()` etc.) so that you get consistent data.